common/persistence/nosql/nosqlplugin/cassandra/gocql/session.go

```diff
@@ -149,7 +157,7 @@ func (s *session) NewBatch(
 func (s *session) ExecuteBatch(
 	b Batch,
 ) (retError error) {
-	defer func() { s.handleError(retError) }()
+	defer func() { s.handleError(b.(*batch).ctx, retError) }()
```

Comment:
You have `context()` there. Why not to propagate it to interface and avoid casting here? Do we really need `Batch` interface at all?

Reply:
I had that originally but then I noticed that the next line does a blind cast to *batch regardless. I felt like I should stay consistent with the approach already taken as much as possible. As for the Batch interface being useful in general ... that's not something I'd want to mess with in the scope of this PR.